src/Std/Time/Internal/UnitVal.lean

```diff
@@ -27,12 +28,33 @@ structure UnitVal (α : Rat) where
   Value inside the UnitVal Value.
   -/
   val : Int
-  deriving Inhabited, BEq
+deriving Inhabited, DecidableEq
+
+@[ext]
+theorem UnitVal.ext {x} {a b : UnitVal x} (h : a.val = b.val) : a = b := by
```

Comment:
Does marking `UnitVal` with `@[ext]` not produce this lemma?

Reply:
I wasn't aware that we have this feature -- yep, this works! I removed my `ext` lemmas.